src/software/ai/navigator/path_manager/path_manager.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+#include <vector>
+
+#include "software/ai/navigator/obstacle/obstacle.h"
+#include "software/ai/navigator/path_planner/path_planner.h"
+#include "software/geom/point.h"
+#include "software/geom/rectangle.h"
+#include "software/geom/spline.h"
+
+/**
+ * PathManager is an interface for a path manager that,
+ * given a list of start and destination point pairs,
+ * a list of static obstacles, and navigable_area will return
+ * a list of 'best' paths between each of the pairs of
+ * start and destination
+ */
+
+using Path = std::optional<Spline>;
+
+// Used to plan a path for a robot
+class PathObjective
+{
+   public:
+    PathObjective(const Point start, const Point end, const double current_velocity,
+                  const std::vector<Obstacle> &avoid_area_obstacles)
+        : start(start),
+          end(end),
+          current_velocity(current_velocity),
+          avoid_area_obstacles(avoid_area_obstacles)
+    {
+    }
+
+    const Point start;
+    const Point end;
+    const double current_velocity;
+    const std::vector<Obstacle> avoid_area_obstacles;
+};
+
+class PathManager
+{
+   public:
+    /**
+     * Returns map of robot ids to paths guided by path objectives,
+     * given navigable_area and static_obstacles
+     *
+     * @param objectives map of robot id to path objectives
+     * @param navigable_area Rectangle representing the navigable area
+     * @param static_obstacles static obstacles to avoid, excluding the robots
+     *  whose paths are being planned
+     *
+     * @return map of robot ids to paths
```

Comment:
Similar comment to above, could we return a map of `robot`->`path`?

Reply:
done